addons/website/static/src/components/configurator/configurator.js

```diff
@@ -200,6 +200,56 @@ class PaletteSelectionScreen extends Component {
         this.logoInputRef.el.click();
     }
 
+    async svgToPng(src) {
+        const updatedSrc = await new Promise(resolve => {
+            const img = new Image();
+            img.onload = () => {
+                // Firefox does not support drawing SVG to canvas unless it has
+                // width and height attributes set on the root <svg>
+                if (img.naturalHeight === 0) {
+                    // Set arbitrary height on image and attach it to the DOM to force width computation.
+                    img.height = 1000;
+                    img.style.opacity = 0;
+                    document.body.appendChild(img);
+
+                    const request = new XMLHttpRequest();
+                    request.open('GET', img.src, true);
+                    request.onload = () => {
+                        // Convert the data URI to a SVG element
+                        const parser = new DOMParser();
+                        const result = parser.parseFromString(request.responseText, 'text/xml');
+                        const SVG = result.getElementsByTagName("svg")[0];
+
+                        // Add the attributes Firefox needs and remove image from the DOM.
+                        SVG.setAttribute('width', img.width);
+                        SVG.setAttribute('height', img.height);
+                        img.remove();
+
+                        // Convert the SVG element to a data URI
+                        const svg64 = btoa(new XMLSerializer().serializeToString(SVG));
+                        resolve('data:image/svg+xml;base64,' + svg64);
+                    };
+                    request.send();
+                } else {
+                    resolve(src);
+                }
+            };
+            img.src = src;
+        });
```

Comment:
@smottet Is not it enough to just do as in the commit I mentioned before? (https://github.com/odoo/odoo/commit/57154071c0df25edc518dbfd87a8e77397ab7e2d) -> use natural width and height and if there is not any, force a width and get the related height? That would be a ~5 LOC update to your original code?

Anyways, I tested this code and it actually does the same as the original one now :shrug: (maybe firefox was updated meanwhile?). However, both do not work on my computer (altough work on runbot). This is probably because of my version of a lib which is not the same as the one on runbot (we tried to fix it with https://github.com/odoo/odoo/pull/68211 but still ongoing...)
If I upload a problematic SVG, I get a palette color from it but then the logo does not appear on my website (broken image :shrug:). In fact, editing its website and choosing those problematic SVG as a logo *is prevented by the editor* on my side ("cannot upload this image blablabla") -> maybe that protection should simply also be there on the configurator side (so I won't have a bug, just the same inability to upload those SVG as via the edito)r:
![image](https://user-images.githubusercontent.com/10338094/129880856-f89bbc91-c840-4c82-8a72-36066131d605.png)

So could you:
- Come back to your original code which do seem to work now, but update it to use naturalWidth and naturalHeight if they exist, otherwise just a forced width (and related height).
- Check it is a supported image before using it (actually, just upload a random non-image file, and your code loads for some seconds then displays a broken stuff... it should just says it is not valid with that message and do nothing else):
![image](https://user-images.githubusercontent.com/10338094/129880856-f89bbc91-c840-4c82-8a72-36066131d605.png)

Reply:
https://stackoverflow.com/questions/28690643/firefox-error-rendering-an-svg-image-to-html5-canvas-with-drawimage

First answer:
> Firefox does not support drawing SVG images to canvas unless the svg file has width/height attributes on the root <svg> element and those width/height attributes are not percentages. This is a [longstanding bug](https://bugzilla.mozilla.org/show_bug.cgi?id=700533).
> 
> You will need to edit the icon.svg file so it meets the above criteria.

Therefore just updating image's width/height doesn't work and we need the extra part where I add width/height directly on the svg...

If I understand the problem you encounters on your computer (everything works well on mine) is due to an incorrect mimetype: `image/svg`? 

Then the solution could be: 
1) keep current code version
2) add a check on upload to make sure mimetype is amongst the authorized one otherwise display error message as you propose in your second bullet point.

Reply:
@qsm-odoo Does the solution I propose in the previous message looks good to you ?

Reply:
1. The less code the better, if SAD's fix was enough at the time, it should be enough now (otherwise your code here should also be used at SAD's fix location too). And it's not much extra code (we force the width to a hardcoded value instead of another one) so no reason not to do it. Your code however is extra code that we cannot prove is necessary... so best not to keep it (unless you have a SVG proving its utility of course).
2. Yes, the upload there should follow the same flow as an upload in the media dialog of the editor.

Reply:
Yes but the SAD's fix didn't need to convert the SVG to a PNG. The extra code comes from the fact we need to convert the SVG to a PNG otherwise `extract_image_primary_secondary_colors` fails because PIL doesn't support SVG format (https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#fully-supported-formats).

I took what was done here for the conversion: https://github.com/odoo/odoo/blob/1632bf17090b38eb8998edb09c1522929dc0312b/addons/website_slides/static/src/js/slides_upload.js#L370-L382

But as I said in a previous message:

> Firefox does not support drawing SVG images to canvas unless the svg file has width/height attributes on the root element and those width/height attributes are not percentages. This is a [longstanding bug](https://bugzilla.mozilla.org/show_bug.cgi?id=700533).
> 
> You will need to edit the icon.svg file so it meets the above criteria.

So this extra code is needed to support the conversion of SVG without width/height attributes...

Reply:
So, if I remove your extra code and use a SVG file without width/height attribute on the root, it will not work, right?

Reply:
Ok, indeed :+1: Maybe worth making an utility method to use it where you copied the code at odoo/addons/website_slides/static/src/js/slides_upload.js ?